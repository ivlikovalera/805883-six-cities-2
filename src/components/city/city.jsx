import React from "react";
import {PropTypes as pt} from 'prop-types';
import {connect} from 'react-redux';
import {getActiveCity} from './../../reducer/data/selector.js';
import {ActionCreator as DataActionCreator} from '../../reducer/data/reducer.js';

export const City = (props) => {
  const {name, onCityClick, activeCity, isFavorite} = props;
  return <a className={`locations__item-link ${isFavorite ? null : checkActiveCity(activeCity.name, name)}`
  } href="#" onClick={(evt) => {
    evt.preventDefault();
    if (!isFavorite) {
      onCityClick(name);
    }
  }
  }>
    <span>{name}</span>
  </a>;
};

const checkActiveCity = (activeName, name) => {
  if (activeName === name) {
    return `tabs__item tabs__item--active`;
  }
  return `tabs__item`;
};

City.propTypes = {
  name: pt.string.isRequired,
  onCityClick: pt.func,
  activeCity: pt.shape({
    name: pt.string,
  }),
  isFavorite: pt.bool,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  activeCity: getActiveCity(state),
});

const mapDispatchToProps = (dispatch) => ({
  onCityClick: (city) => {
    dispatch(DataActionCreator.changeCity(city));
    dispatch(DataActionCreator.getOffers());
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(City);
