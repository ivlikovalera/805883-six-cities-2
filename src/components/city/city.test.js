import React from 'react';
import renderer from 'react-test-renderer';
import {City} from './city.jsx';

it(`City correctly renders after relaunch`, () => {

  const cityComponent = renderer
  .create(<City
    name={``}
    onCityClick={() => {}}
    activeCity={{}}
    isFavorite={true || false}
  />)
  .toJSON();

  expect(cityComponent).toMatchSnapshot();
});
