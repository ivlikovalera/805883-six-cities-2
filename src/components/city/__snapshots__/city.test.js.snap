// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City correctly renders after relaunch 1`] = `
<a
  className="locations__item-link null"
  href="#"
  onClick={[Function]}
>
  <span>
    
  </span>
</a>
`;
